### ISSUE TITLE
Update s3://commoncrawl/ access scheme

### ISSUE THREAD

**Issue by sebastian-nagel** [fhamborg/news-please#223]
Hi @fhamborg,

next week the scheme how data on s3://commoncrawl/ is accessed will change:
- `https://commoncrawl.s3.amazonaws.com/` should be replaced by `https://data.commoncrawl.org/`
- WARC file listings via `aws s3 ls` require authentication
  - for AWS users: remove `--no-sign-request`
  - otherwise WARC file listings for the news crawl are provided on `https://data.commoncrawl.org/crawl-data/CC-NEWS/year/month/warc.paths.gz`, eg.  https://data.commoncrawl.org/crawl-data/CC-NEWS/2022/04/warc.paths.gz

For further information, please see:
- https://commoncrawl.org/2022/03/introducing-cloudfront-access-to-common-crawl-data/
- https://commoncrawl.org/access-the-data/
- https://data.commoncrawl.org/crawl-data/CC-NEWS/index.html

If required I might be able to contribute a patch.

**Comment by fhamborg** [comment]
Hi @sebastian-nagel , thanks for posting this! After reading https://commoncrawl.org/access-the-data/, am I right that only to get the list of files we need to change the code? But the download code can stay as is?

**Comment by fhamborg** [comment]
FYI, the listing code is here: https://github.com/fhamborg/news-please/blob/master/newsplease/crawler/commoncrawl_crawler.py#L166=

And the download code here: https://github.com/fhamborg/news-please/blob/a66e9486f1da36950bfc913113c408f66efd49fc/newsplease/crawler/commoncrawl_extractor.py#L230=

**Comment by sebastian-nagel** [comment]
Yes, the download can stay the same but [__cc_base_url](https://github.com/fhamborg/news-please/blob/a66e9486f1da36950bfc913113c408f66efd49fc/newsplease/crawler/commoncrawl_extractor.py#L55) should point to `https://data.commoncrawl.org/`.

Regarding, the listing code: so far, only listings by month are provided.

**Comment by fhamborg** [comment]
Alright, so I hope my small "fix" addressed this changes you described. I'm currently having issues with uploading news-please to pypi but once these are addressed, I will be able to publish a new version that fixes this.

**Comment by joemkwon** [comment]
BTW the commoncrawl_crawler.py code is not updated, so the code doesn't work as is yet. Lines 163 and 166 need to be edited to either wget with https, or removing no-sign-request and having authenticated AWS setup I think.

**Comment by sshleifer** [comment]
I have an aws access key, and put it into `aws configure`, but
```
aws s3 ls --recursive s3://commoncrawl/crawl-data/CC-NEWS/                  
```
still fails with
```
An error occurred (InvalidAccessKeyId) when calling the ListObjectsV2 operation: The AWS Access Key Id you provided does not exist in our records.
```

Does someone have a working replacement for the old:

```bash
aws s3 ls --recursive s3://commoncrawl/crawl-data/CC-NEWS/ --no-sign-request | awk '{ print $4 }' 
```

**Comment by sebastian-nagel** [comment]
Is there is a typo in the access key - stored in ~/.aws/credentials?

The following shell script generates a list of all CC-NEWS WARC files without the need to own an AWS account:
```bash
#!/bin/bash

for year in $(seq 2016 $(date +%Y)); do
    for month in $(seq 1 12); do
        yearmonth=$year/$(printf "%02d" $month)
        if [[ $yearmonth < 2016-08 ]]; then
            : # news crawl started August 2016
        elif [[ $yearmonth > $(date +%Y/%m) ]]; then
            : # after current month
        else
            curl https://data.commoncrawl.org/crawl-data/CC-NEWS/$yearmonth/warc.paths.gz \
                 | gzip -dc
        fi
    done
done
```

But for news-please this should be rewritten using pure Python, eventually using a dual approach (boto3 for AWS users). I'll try to provide a PR.

**Comment by fhamborg** [comment]
@sebastian-nagel thanks a lot, I would definitely appreciate a PR :)

**Comment by sshleifer** [comment]
Yes, I screwed up the access key. Sorry and thank you @sebastian-nagel

**Comment by joemkwon** [comment]
@sebastian-nagel have you been able to get the script functioning since the new access scheme? I created and configured an AWS account, and since scraping the recent months, have found that about a third of the articles just have an empty JSON string instead of the usual information. This seems unusually high and I only changed a few lines in the scraper since things have changed. Have you noticed anything similar?
